astropy/io/misc/parquet.py

```diff
@@ -513,3 +513,149 @@ def get_pyarrow():
         writer_version = "2.0"
 
     return pa, parquet, writer_version
+
+
+def write_parquet_votable(table, output, *, metadata, overwrite=False):
+    """
+    Writes a Parquet file with a VOT (XML) metadata table included.
+
+    Parameters
+    ----------
+    table : `~astropy.table.Table`
+        Data table that is to be written to output.
+    output : str or path-like
+        The filename to write the table to.
+    metadata : dict
```

Comment:
Is the user expected to generate this metadata object?  How does it relate to the `.meta` info that may already be in the `table` or its columns?

Reply:
I have issues with this kwarg, too, and wrote up a few test cases. E.g. I think we should reuse the column metadata if the Table already has it (e.g. the table could have been read in from a votable/be a result of an astroquery query). So I would instead add another kwarg for `metadata_overwrite`, and then use the `metadata` kward to overwrite some of the column metadata (but still, it would be nice to allow partial overwrite).

As far as I see, we do nothing with the table metadata, but that is already the case when reading in from a votable? (E.g. with this gaia file, it's not clear how to get the e.g. `QUERY` info out of it, etc.)

Reply:
in `votable.parquet` we call this `column_metadata`, so I suppose we need to do the same. And it's similarly a mandatory kwarg, which IMO needs to be changed and allowed it to use existing metadata if/when it's sufficiently provided.